controllers/apidefinition_controller.go

```diff
@@ -144,74 +144,76 @@ func (r *ApiDefinitionReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		}
 
 		// Check GraphQL Federation
-		if upstreamRequestStruct.Spec.GraphQL != nil && upstreamRequestStruct.Spec.GraphQL.GraphRef != "" {
-			if upstreamRequestStruct.Spec.GraphQL.ExecutionMode == model.SubGraphExecutionMode {
-				subgraph := &tykv1alpha1.SubGraph{}
-
-				err := r.Client.Get(ctx, types.NamespacedName{
-					Namespace: req.Namespace,
-					Name:      desired.Spec.GraphQL.GraphRef,
-				}, subgraph)
-				if err != nil {
-					return err
-				}
-
-				upstreamRequestStruct.Spec.GraphQL.Schema = subgraph.Spec.Schema
-				upstreamRequestStruct.Spec.GraphQL.Subgraph.SDL = subgraph.Spec.SDL
+		if upstreamRequestStruct.Spec.GraphQL != nil {
```

Comment:
Just FYI, this code block will be completely refactored with #522
If new UDG does not require changes in this code block, please let's wait #522 to prevent possible conflicts.